src/wallet/spend.cpp

```diff
@@ -468,6 +498,19 @@ CoinsResult AvailableCoins(const CWallet& wallet,
         }
     }
 
+    if (params.check_version_trucness && unconfirmed_truc_coins.size() > 0) {
+        auto highest_value_truc_tx = std::max_element(truc_txid_by_value.begin(), truc_txid_by_value.end(), [](const auto& tx1, const auto& tx2){
```

Comment:
Do I understand correctly that the approach here is to just return 1 unconfirmed TRUC coin from `AvailableCoins`, i.e. the one with the highest value? That seems like it could weaken coin selection quality a little bit - is there a way to encode these kinds of restrictions (once coin X is selected, ABC are off the table) into the coin selection process instead?

This is probably rare in practice (I think people would only use unconfirmed TRUC when CPFPing or really desperate), but it would be helpful to provide justification here.